mono/metadata/exception.c

```diff
@@ -1491,11 +1497,11 @@ mono_error_set_simple_file_not_found (MonoError *error, const char *file_name, g
 }
 
 void
-mono_error_set_argument_out_of_range (MonoError *error, const char *name)
+mono_error_set_argument_out_of_range (MonoError *error, const char *param_name)
 {
-	ERROR_DECL (error_creating_exception);
-	mono_error_set_exception_handle (error, mono_new_exception_argument_out_of_range (name, error_creating_exception));
-	mono_error_cleanup (error_creating_exception);
+	mono_error_set_specific (error, MONO_ERROR_ARGUMENT_OUT_OF_RANGE, "MonoArgumentException:%s", param_name);
```

Comment:
Something fishy here. You pass param_name for %s but then check it for NULL.

Reply:
Yeah I just realized that lol.
But here is the question, should I convert it by myself or add a "const char* msg" into the parameter so that the use of function would be more consistent to other argument related function?

Reply:
or just leave it alone?

Reply:
Like this? If it is not good then we can just revert it.